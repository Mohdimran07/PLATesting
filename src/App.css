.App{
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-align: center;
}

.heading{
  background-color: rgb(158, 42, 158);
  width: 100%;
  margin-bottom: 10px;
  color: white;
}