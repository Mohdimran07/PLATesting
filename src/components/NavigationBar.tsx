import { AppBar,  makeStyles, Toolbar, Typography } from "@material-ui/core";
import { ClassNameMap } from "@material-ui/core/styles/withStyles";
import React from "react";

const useStyles = makeStyles(() => ({
    root: {
        fontSize: "40px"
    }
}))

const NavigationBar = (): JSX.Element => {
    const classes: ClassNameMap = useStyles();
    return (
        <AppBar position="static">
            <Toolbar>
                <Typography className={classes.root}>Post List App</Typography>
            </Toolbar>
        </AppBar>
    )
}

export default NavigationBar;