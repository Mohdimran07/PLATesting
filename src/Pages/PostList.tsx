import { Box, CircularProgress, TableContainer, Table, TableHead, TableCell, TableBody, TableRow, Button, Pagination } from "@mui/material";
import { color } from "@mui/system";
import React,{ChangeEvent} from "react";
import { NavigateFunction, useNavigate } from "react-router-dom";

import { postDataType } from "../App";

type listOfPost = {
  postData: postDataType;
  selectedPage: number;
  setSelectedPage: React.Dispatch<React.SetStateAction<number>>;
}

type dataType = {
  author: string;
  created_at: string;
  title: string;
  url: string;
  objectID: string;
}

const PostList = ({postData, selectedPage, setSelectedPage}: listOfPost): JSX.Element => {
   const navigate: NavigateFunction = useNavigate();

  const selectHandler = (jsonData: dataType): void => {
    navigate("rawdata", { state: jsonData})
  }
  return (
    <Box 
      sx={{
        display: "flex",
        flexDirection: "column",
        justifyContent: "center",
        alignItems: "center",
        marginTop:"20px"
      }}
    >
       {postData.length === 0 ? (
         <>
         <CircularProgress></CircularProgress> Loading
         </>
       ): (
         <Box 
          sx={{
            display: "flex",
            flexDirection: "column",
            justifyContent:"center",
            alignItems: "center"
          }}
         >
          <TableContainer sx={{ maxHeight: 500 }}>
            <Table stickyHeader>
              <TableHead>
                <TableCell 
                  sx={{
                    backgroundColor: "black",
                    border: "2px solid red",
                    color: "white"
                  }} align="center"
                >
                  Title
                </TableCell>
                <TableCell 
                  sx={{
                    backgroundColor: "black",
                    border: "2px solid red",
                    color: "white"
                  }} align="center"
                >
                  Author
                </TableCell>
                <TableCell 
                  sx={{
                    backgroundColor: "black",
                    border: "2px solid red",
                    color: "white"
                  }} align="center"
                >
                  Created-At
                </TableCell>
                <TableCell 
                  sx={{
                    backgroundColor: "black",
                    border: "2px solid red",
                    color: "white"
                  }} align="center"
                >
                  URL
                </TableCell>
                <TableCell 
                  sx={{
                    backgroundColor: "black",
                    border: "2px solid red",
                    color: "white"
                  }} align="center"
                >
                  Raw-Data
                </TableCell>
              </TableHead>
              <TableBody>
                {postData.length && postData[selectedPage - 1].map((items: dataType) => {
                  console.log(items)
                  const {author, created_at, title, url, objectID} = items;
                  return(
                    <TableRow key={objectID}>
                      <TableCell sx={{ maxWidth:250, overflow: "hidden"}}>{title}</TableCell>
                      <TableCell>{author}</TableCell>
                      <TableCell>{created_at}</TableCell>
                      <TableCell sx={{ maxWidth: 200, overflow: "hidden"}}>{url}</TableCell>
                      <TableCell>
                        <Button variant="contained" onClick={(): void => selectHandler(items)}>Select</Button>
                      </TableCell>
                    </TableRow>
                  )
                })}
              </TableBody>
            </Table>
          </TableContainer>
          <Box sx={{padding: "50px"}}>
            <Pagination count={postData.length} 
            onChange={(e: ChangeEvent<any>, value: number): void => setSelectedPage(value)}
            page={selectedPage}></Pagination>
          </Box>
         </Box>
       )}
    </Box>
  )
}

export default PostList;