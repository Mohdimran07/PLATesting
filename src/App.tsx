import React, { useEffect, useMemo, useState } from "react";
import "./App.css";
import { Routes, Route } from "react-router-dom";
import RawData from "./Pages/RawJson";
import NotFound from "./Pages/NotFound";
import PostList from "./Pages/PostList";
import axios from "axios";

import NavigationBar from "./components/NavigationBar";

export type postDataType = any[];

function App(): JSX.Element {
  const [postData, setPostData] = useState<postDataType>([]);
  const [selectedPage, setSelectedPage] = useState<number>(1);

  let count: number = 0;

  const fetchApi = async (): Promise<void> => {
    const response = await axios.get(
      `http://hn.algolia.com/api/v1/search?query=bar&page=${count}`
    );

    if (response) {
      console.log(response);
      setPostData(
        (prev: postDataType): postDataType => [...prev, response?.data?.hits]
      );
    }

    count += 1;
  };

  useEffect(() => {
    fetchApi();
    setInterval(fetchApi, 10000);
  }, []);

  useMemo(() => {
    console.log(postData);
  }, [postData]);

  return (
    <div className="App">
      <NavigationBar />
      <Routes>
        <Route
          path="/"
          element={
            <PostList
              postData={postData}
              selectedPage={selectedPage}
              setSelectedPage={setSelectedPage}
            ></PostList>
          }
        ></Route>
        <Route path="rawdata" element={<RawData></RawData>}></Route>
        <Route path="*" element={<NotFound></NotFound>}></Route>
      </Routes>
    </div>
  );
}

export default App;
