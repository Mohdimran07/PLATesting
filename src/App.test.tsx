import React from "react";
import { render, screen, act } from "@testing-library/react";
import App from "./App";
import { BrowserRouter } from "react-router-dom";
import NavigationBar from "./components/NavigationBar";
import PostList from "./Pages/PostList";
import axios from "axios";

test("renders App component", () => {
  render(
    <BrowserRouter>
      <App />
    </BrowserRouter>
  );
  screen.debug();
});
test("SnapShot testing ", () => {
  const { container } = render(
    <BrowserRouter>
      <App />
    </BrowserRouter>
  );
  expect(container.firstChild).toMatchSnapshot();
});

describe("testing on Nav Bar", () => {
  test("Is NavBar component exist", () => {
    render(
      <BrowserRouter>
        <NavigationBar></NavigationBar>
      </BrowserRouter>
    );

    screen.debug();
  });

  test("Does NavBar component have text?", () => {
    render(
      <BrowserRouter>
        <NavigationBar></NavigationBar>
      </BrowserRouter>
    );
    const textElement = screen.getByText(/Post/);
    expect(textElement).toBeInTheDocument();
  });

  test("How many banner exist in NavBar", () => {
    render(
      <BrowserRouter>
        <NavigationBar></NavigationBar>
      </BrowserRouter>
    );
    const banner = screen.getAllByRole("banner");
    expect(banner.length).toBe(1);
  });

  test("snapshot testing", () => {
    const { container } = render(
      <BrowserRouter>
        <NavigationBar></NavigationBar>
      </BrowserRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

const DummyData = {
  author: "schoen",
  created_at: "2018-02-07T22:22:28.000Z",
  title: "John Perry Barlow has died",
  url: "https://www.eff.org/deeplinks/2018/02/john-perry-barlow-internet-pioneer-1947-2018",
  objectID: "1234",
};

describe("rendering PostList App", () => {
  test("PostList component", () => {
    const listProps = {
      postData: DummyData,
      selectedPage: 1,
      setSelectedPage: 1,
    };
    render(
      <BrowserRouter>
        <PostList {...listProps} />
      </BrowserRouter>
    );
  });
  screen.debug();
});

jest.mock("axios");

// describe(" Integration testing of axios", () => {
//   test(" testing on axios", async () => {
//     const resolvedPromise = Promise.resolve({ data: { hits: DummyData } });
//     axios.get.mockImplementation(() => resolvedPromise);
//     render(
//       <BrowserRouter>
//         <App />
//       </BrowserRouter>
//     );
//     screen.debug();
//   });
// });
