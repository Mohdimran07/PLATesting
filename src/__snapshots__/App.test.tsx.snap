// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapShot testing  1`] = `
<div
  class="App"
>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <p
        class="MuiTypography-root makeStyles-root-2 MuiTypography-body1"
      >
        Post List App
      </p>
    </div>
  </header>
  <div
    class="MuiBox-root css-1tkyuqa"
  >
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
     Loading
  </div>
</div>
`;

exports[`renders App component 1`] = `
<div
  class="App"
>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <p
        class="MuiTypography-root makeStyles-root-2 MuiTypography-body1"
      >
        Post List App
      </p>
    </div>
  </header>
  <div
    class="MuiBox-root css-1tkyuqa"
  >
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
     Loading
  </div>
</div>
`;

exports[`testing on Nav Bar snapshot testing 1`] = `
<header
  class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
>
  <div
    class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <p
      class="MuiTypography-root makeStyles-root-6 MuiTypography-body1"
    >
      Post List App
    </p>
  </div>
</header>
`;
